modules/lib/zsh.nix

```diff
@@ -0,0 +1,28 @@
+{ lib }:
+
+rec {
+  # Produces a Zsh shell like value
+  toZshValue = v: if builtins.isBool v then
+                    if v then "true" else "false"
+                  else if builtins.isString v then
+                    "\"${v}\""
+                  else if builtins.isList v then
+                    "(${lib.concatStringsSep " " (map toZshValue v)})"
+                  else "\"${toString v}\"";
+
+  # Produces a Zsh shell like export statement
+  export = n: v: "export ${n}=${toZshValue v}";
```

Comment:
This could probably be changed to something like this:
```
export = n: v: "export ${define n v}";
```

Reply:
Good idea. Changed that.